### PR TITLE
feat: Spinner now use Icon

### PR DESCRIPTION
**Pull request by Crash--**
should fix https://github.com/cozy/cozy-ui/issues/878 

https://crash--.github.io/cozy-ui/react/#spinner

### CODE REVIEW

File: react/Spinner/index.jsx [hunk 1]
```diff
@@ -13,19 +14,44 @@ export const Spinner = ({
...
-          [styles[`c-spinner--${size}`]]: size
+          [styles['c-spinner--nomargin']]: noMargin
         },
         className
       )}
     >
+      <Icon
+        icon="spinner"
+        color={color}
+        className={iconStyle['icon--spin']}
```
Comment (GoOz) [inline react/Spinner/index.jsx]: `<Icon />` has a props `spin` that already add the class.
See https://github.com/cozy/cozy-ui/blob/master/react/Icon/index.jsx#L45-L47
Unless you had an issue with this but it should use the props, right?

Comment (Crash--) [reply react/Spinner/index.jsx]: thanls!

File: react/Spinner/index.jsx [hunk 2]
```diff
@@ -13,19 +13,39 @@ export const Spinner = ({
   size,
   className
 }) => {
+  let realSize = ''
+  switch (size) {
```
Comment (cedricmessiant) [inline react/Spinner/index.jsx]: why not a mapping?

```
const realsizeMapping = {
  tiny: 8,
  small: 12,
  ...
}

const realsize = realsizeMapping(size)```

File: react/Spinner/Readme.md [hunk 3]
```diff
@@ -38,12 +38,21 @@ Show a spinner when loading something.
...
+
+```
+const I18n = require('../I18n').default;
+
+  <div>
+    <I18n lang='en' dictRequire={() => {}}>
+      <Spinner size='xxlarge' loadingType='hi' />
+    </I18n>
+
+</div>
```
Comment (cedricmessiant) [inline react/Spinner/Readme.md]: maybe this doc change should be in a distinct commit?

**Comment by y-lohse** [comment]
@GoOz do you know if these variables are still useful ? https://github.com/cozy/cozy-ui/blob/c3232993d654b818618990c7e9e04b53d641d8f6/stylus/settings/icons.styl#L52-L75 They seem unused in cozy-ui. How likely is it that they are used in apps?

**Comment by Crash--** [comment]
> They seem unused in cozy-ui. How likely is it that they are used in apps? 

In fact some of them are actually used : `$icon-16` for instance. 

Also I wanted to remove `$icon-spinner-white` (and others with the associated svg) but they are used for the button

**Comment by GoOz** [comment]
I guess apps that use the old spinner way have a chance to use those placeholders. Not sure which one though.

**Pull request merged**